src/backend/backend.ts

```diff
@@ -220,89 +205,84 @@ export default class Backend {
-    console.info('Get number of files...', criteria, matchAny);
-    return this.fileRepository.count(
-      criteria
-        ? {
-            criteria,
-            matchAny,
-          }
-        : undefined,
-    );
+  async countFiles(): Promise<[fileCount: number, untaggedFileCount: number]> {
+    console.info('Get number stats of files...');
+    const fileCount = this.fileRepository.count();
+    const untaggedFileCount = this.fileRepository.countExact({
+      key: 'tags',
+      operator: 'contains',
+      value: [],
+      valueType: 'array',
+    });
+    return Promise.all([fileCount, untaggedFileCount]);
   }
 
   // Creates many files at once, and checks for duplicates in the path they are in
-  async createFilesFromPath(path: string, files: IFile[]): Promise<IFile[]> {
+  async createFilesFromPath(path: string, files: FileDTO[]): Promise<void> {
     console.info('Backend: Creating files...', path, files);
     // Search for file paths that start with 'path', so those can be filtered out
-    const criteria: IStringSearchCriteria<IFile> = {
+    const existingFilesInPath = await this.fileRepository.findExact({
       valueType: 'string',
       operator: 'startsWith',
       key: 'absolutePath',
       value: path,
-    };
-    const existingFilesInPath: IFile[] = await this.fileRepository.find({ criteria });
+    });
     console.debug('Filtering files...');
     const newFiles = files.filter((file) =>
       existingFilesInPath.every((f) => f.absolutePath !== file.absolutePath),
     );
     console.debug('Creating files...');
-    const res = await this.fileRepository.createMany(newFiles);
+    await this.fileRepository.createMany(newFiles);
     console.debug('Done!');
-    return res;
   }
 
-  async clearDatabase(): Promise<void> {
+  async clear(): Promise<void> {
     console.info('Clearing database...');
     return dbDelete(DB_NAME);
   }
 
-  async backupDatabaseToFile(path: string): Promise<void> {
+  async backupToFile(path: string): Promise<void> {
     const blob = await exportDB(this.db, { prettyJson: false });
     // might be nice to zip it and encode as base64 to save space. Keeping it simple for now
     await fse.ensureFile(path);
     await fse.writeFile(path, await blob.text());
   }
 
-  async restoreDatabaseFromFile(path: string): Promise<void> {
+  async restoreFromFile(path: string): Promise<void> {
     const buffer = await fse.readFile(path);
     const blob = new Blob([buffer]);
-    await this.clearDatabase();
+    await this.clear();
     console.log('Importing database backup', path);
     await importDB(blob);
     // There also is "importInto" which as an "clearTablesBeforeImport" option,
     // but that didn't seem to work correctly (files were always re-created after restarting for some reason)
   }
 
-  async peekDatabaseFile(path: string): Promise<{ numTags: number; numFiles: number }> {
+  async peekFile(path: string): Promise<[numTags: number, numFiles: number]> {
```

Comment:
neat, I've seen them around, but didn't realise you could do named array values like this.